Jira (#1145)
returntocorp/semgrep-docs: docs/semgrep-cloud-platform/jira.md

* Jira can only be enabled through Semgrep Cloud Platform (SCP). [Create an account](/semgrep-code/getting-started/#signing-in-to-semgrep-cloud-platform) to set up your Jira integration.
* To create tickets, you must [add or onboard the project](/semgrep-code/getting-started/#option-b-adding-a-repository-from-github-gitlab-or-bitbucket) (repository) to Semgrep Cloud Platform for scanning.
:::

To integrate Jira, follow these steps:

1. In [Semgrep Cloud Platform](https://semgrep.dev/login), go to **Settings** > **[Integrations](https://semgrep.dev/orgs/-/settings/integrations)**.
2. On the **[Integrations](https://semgrep.dev/orgs/-/settings/integrations)** page, click **Add Integration** (or **Setup First Integration** if this is your first integration), and then select **Jira**.
3. Follow the on-screen instructions to add your Jira API key.

## Configuring your default project and ticket type

Use the drop-downs to select the default project and ticket type. These settings can be changed later from the integrations page.

![Jira configuration modal](/img/jira-configure-defaults.png) 

**Figure** A sample Jira integration

## Creating tickets

### Code

:::note

> Creating tickets for many findings at once may take some time. Tickets that take longers than 10 seconds to create will be shown in the UI once the page is refreshed.

Reviewer: (Not a suggestion - just a note - the "will" here totally makes sense and doesn't have to be changed.)

:::

To create tickets for one or more Code findings, use the triage button from the Code findings page or from an individual finding's details page. Findings can be individually selected, or you can create tickets for all findings for a given rule. 

Once a ticket has been created, a link will be present on the right side of the findings page and along the top of an individual finding's details page.

![Create Jira ticket - Code](/img/jira-code-findings.png)
**Figure** Code triage flow

![Jira ticket created - Code](/img/jira-code-ticketed.png)
**Figure** Code ticket created

### Supply Chain

> Tickets can only be created for Supply Chain findings with reachable usages. They cannot be created for "Always reachable" or "Unreachable" findings.

Reviewer: ```suggestion
Tickets can only be created for Supply Chain findings with reachable usages. They cannot be created for **Always reachable** or **Unreachable" findings**.
```

The Supply Chain triage flow will create tickets for all instances of a given vulnerability. Instances can not be selected individually. 

![Create Jira ticket - Supply Chain](/img/jira-ssc-findings.png)
**Figure** Supply Chain triage flow

![Jira ticket created - Supply Chain](/img/jira-ssc-ticketed.png)
**Figure** Supply Chain ticket created

## Removing your Jira integration

On the **[Integrations](https://semgrep.dev/orgs/-/settings/integrations)** page, find Jira, click **Edit Configuration** and the **Delete**

> * This **will not** delete any tickets created by Semgrep

Reviewer: ```suggestion
* This **does not** delete any tickets created by Semgrep.
```
"In general, use present tense rather than future tense; in particular, try to avoid using will where possible." ([Google Style Guide - Tenses](https://developers.google.com/style/tense))

> * This **will remove** the link between Jira tickets and Semgrep findings, even if the integration is re-added in the future

Reviewer: ```suggestion
* This **removes** the link between Jira tickets and Semgrep findings, even if the integration is re-added in the future.
```
Google Style Guide - Tenses, same as previous comment.
